src/cpp/DataBrokerLocalParticipant.cpp

```diff
@@ -30,6 +33,45 @@ DataBrokerLocalParticipant::DataBrokerLocalParticipant(
 {
 }
 
+eprosima::fastrtps::rtps::GuidPrefix_t DataBrokerLocalParticipant::guid()
+{
+    if (configuration_.discovery_protocol == fastrtps::rtps::DiscoveryProtocol::SERVER)
+    {
+        return configuration_.server_guid;
+    }
+    else
+    {
+        return DataBrokerParticipant::guid();
+    }
+}
+
+eprosima::fastdds::dds::DomainParticipantQos DataBrokerLocalParticipant::participant_qos()
+{
+    eprosima::fastdds::dds::DomainParticipantQos pqos = DataBrokerParticipant::participant_qos();
+
+    if (configuration_.discovery_protocol ==
+            fastrtps::rtps::DiscoveryProtocol::SERVER)
+    {
+        // Set this participant as a SERVER
+        pqos.wire_protocol().builtin.discovery_config.discoveryProtocol =
+                fastrtps::rtps::DiscoveryProtocol::SERVER;
```

Comment:
You have the variable in the configuration and you don't even use it :crying_cat_face: